include/openthread/error.h

```diff
@@ -234,6 +234,11 @@ typedef enum OT_MUST_USE_RESULT otError
      */
     OT_ERROR_PENDING = 36,
 
+    /**
+     * Operation is cancelled becuase RCP is recovering from failures.
+     */
+    OT_ERROR_RCP_RECOVERING = 37,
```

Comment:
In general `otError` is for errors used by OT core modules. Core should not be aware of RCP so having error specific for this and the recovering process of RCP does not sound good to me.

Some Ideas/suggestions:
- Can you use existing error codes, like `BUSY`, `FAILED`, (maybe `PENDING`) for this purpose
- Looking at the use of this, seems like it's used to track whether the RCP recovery is happening to decide to save/discard frames. Can a boolean flag variable or state variable used to track this instead for this purpose instead of a new error value?

--------

(BTW,  I have not had the chance to read the rest of code yet, just noticed this `otError` addition so added this suggestion).

Reply:
Removed this error type.